k9mail/src/test/java/com/fsck/k9/message/html/HtmlConverterTest.java

```diff
@@ -207,4 +207,110 @@ public void testLinkifyBitcoinAndHttpUri() {
                 "http://example.com/" +
                 "</a>", outputBuffer.toString());
     }
+
+    @Test
+    public void testIssue2259Spec() {
+        String text = "text\n" +
+                "---------------------------\n" +
+                "some other text\n" +
+                "===========================\n" +
+                "more text\n" +
+                "-=-=-=-=-=-=-=-=-=-=-=-=-=-\n" +
+                "scissors below\n" +
+                "-- >8 --\n" +
+                "other direction\n" +
+                "-- 8< --\n" +
+                "end";
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">text<hr />" +
+                "some other text<hr />" +
+                "more text<hr />" +
+                "scissors below<hr />" +
+                "other direction<hr />" +
+                "end</pre>",
+                result);
+    }
+
+    @Test
+    public void testDontTouchDashesContainingExcessSpace() {
+        String text = "hello\n--- --- --- --- ---   \nfoo bar";
+        //                                       ^^^ Only 2 spaces allowed in hr dot-dash.
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">hello<br />--- --- --- --- ---   <br />foo bar</pre>",
+                result);
+    }
+
+    @Test
+    public void testConsumeDblSpacesWhenHRsSurround() {
+        String text = "hello\n--- --- --- --- ---   --- --- --- ---\nfoo bar";
+        //                                       ^^^ Because another sequence follows, merge
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">hello<hr />foo bar</pre>", result);
+    }
+
+    @Test
+    public void testMergeConsecutiveBreaksIntoOne() {
+        String text = "hello\n---  ---  ---  --- \n--------- --- ---\nfoo bar";
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">hello<hr />foo bar</pre>", result);
+    }
+
+    @Test
+    public void testIgnoreDashedHorizontalRulePrefixedWithText() {
+        String text = "hello----\n\n";
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">hello----<br /><br /></pre>", result);
+    }
+
+    @Test
+    public void testPreserveUnitaryDecrement() {
+        String text = "for (int i = my.length; i > 0; i--);\n\n";
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">for (int i = my.length; i &gt; 0; i--);<br /><br /></pre>",
+                result);
+    }
+
+    @Test
+    public void testPreserveAnyDoubletOfHypenAlikes() {
```

Comment:
Please make these separate tests. If this one fails it's not immediately obvious why exactly.